### PR TITLE
MF2 Improvements and Fixes Mark 3

### PR DESCRIPTION
**Pull request by dshanske**
This correctly identifies all the test data and adds some additional enhancements.

### CODE REVIEW

**Comment by pfefferle** [comment]
OK, the tests seems to be green, but I still have problems if I try some mentions by hand.

**Comment by pfefferle** [comment]
I hate mf2 parsing!

File: includes/class-linkbacks-mf2-handler.php [hunk 1]
```diff
@@ -127,75 +130,213 @@ public static function generate_commentdata( $commentdata ) {
...
+	 * Is string a URL.
+	 *
+	 * @param array $string
+	 * @return bool
+	*/
+	public static function is_url( $string ) {
+		if ( ! is_string( $string ) ) {
+			return false;
+		}
+		return preg_match( '/^https?:\/\/.+\..+$/', $string );
```
Comment (pfefferle) [inline includes/class-linkbacks-mf2-handler.php]: can you add the possibility to test with local urls? perhaps it is ok to remove the check of the `dot`

File: includes/class-mf2-debugger.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<?php
```
Comment (pfefferle) [inline includes/class-mf2-debugger.php]: Can you please verify WordPress coding style?

**Comment by dshanske** [comment]
I may change to a less stringent URL verification.

**Comment by dshanske** [comment]
I looked at the two failed tests. Because no URL in any of the h-cards matches the URL of the submitted page, it fails. The example output shows an author name that shouldn't be parsed. 

Either way, I changed the is_url function to be less restrictive if WP_DEBUG is on and more if it isn't.

File: includes/class-mf2-debugger.php (hunk 2 above)
Comment (dshanske) [inline includes/class-mf2-debugger.php]: As far as I can see, it is verified, except for the MF2 Parser, which isn't under my control.

**Comment by pfefferle** [comment]
now the tests are failing! do you think it's because of the author changes?

**Comment by dshanske** [comment]
The author test is failing because of the issue as mentioned. But it is my opinion that it is correct. The comment author in the test is not correct.

**Comment by dshanske** [comment]
Unless the test forces the source URL to be on the same domain as the source author URL, those tests fail.

**Comment by pfefferle** [comment]
can you fix the test?

**Comment by dshanske** [comment]
Having bad luck with the tests.But it works in real life.

**Comment by pfefferle** [comment]
This does not help... can you please fix tests https://travis-ci.org/pfefferle/wordpress-semantic-linkbacks

**Comment by dshanske** [comment]
This may take a bit. But will work on it.

**Comment by pfefferle** [comment]
thanks. If the tests does not work, we have a really hard time to add changes.

**Pull request merged**